user-bag-service/src/test/kotlin/io/baum/userbagservice/users/web/UserControllerTest.kt

```diff
@@ -0,0 +1,120 @@
+package io.baum.userbagservice.users.web
+
+import arrow.core.Either
+import io.baum.userbagservice.users.tools.hideSensitiveFields
+import io.baum.userbagservice.users.web.client.ApiClient
+import io.baum.userbagservice.users.web.client.ApiClientFactory
+import io.baum.userbagservice.users.web.model.RemoteAddress
+import io.baum.userbagservice.users.web.model.UserModel
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class UserControllerTest {
```

Comment:
It looks like this tests the code works which is great, how do you test your endpoint is set up right? e.g. that you've got the right path params + query params + routes + http methods etc?

Reply:
I believe integration testing would be key for checking the aforementioned qualities. Was considering in the future maybe setting up a BDD integration testing suite.

Reply:
Have a read into Spring Controller Testing

https://spring.io/guides/gs/testing-web/